nodes/generator/mesh_eval.py

```diff
@@ -0,0 +1,269 @@
+    
+    def validate_json(self, json):
+        if not "vertices" in json:
+            raise Exception("JSON does not have `vertices' key")
+        if not "edges" in json:
+            raise Exception("JSON does not have `edges' key")
+        if not "faces" in json:
+            raise Exception("JSON does not have `faces' key")
+
+    def get_variables(self):
+        variables = set()
+        json = self.load_json()
+        if not json:
+            return variables
+
+        for vertex in json["vertices"]:
+            for c in vertex:
+                if isinstance(c, str):
+                    vs = get_variables(c)
+                    variables.update(vs)
+
+        return list(sorted(list(variables)))
+
+    def adjust_inputs(self):
+        variables = self.get_variables()
+        #print("adjust_input:" + str(variables))
+        #print("inputs:" + str(self.inputs.keys()))
+        for key in self.inputs.keys():
+            if key not in variables:
+                print("Input {} not in variables {}, remove it".format(key, str(variables)))
+                self.inputs.remove(self.inputs[key])
+        for v in variables:
+            if v not in self.inputs:
+                print("Variable {} not in inputs {}, add it".format(v, str(self.inputs.keys())))
+                self.inputs.new('StringsSocket', v)
+
+    def update(self):
+        '''
+        update analyzes the state of the node and returns if the criteria to start processing
+        are not met.
+        '''
+
+        # keeping the file internal for now.
+        if not (self.filename in bpy.data.texts):
+            return
+
+        if not ('Edges' in self.outputs):
+            return
+
+        self.adjust_inputs()
+
+    def get_input(self):
+        variables = self.get_variables()
+        result = {}
+        for var in variables:
+            if var in self.inputs and self.inputs[var].is_linked:
+                result[var] = self.inputs[var].sv_get()[0]
+            else:
+                result[var] = [0.0]
+            #print("get_input: {} => {}".format(var, result[var]))
+        return result
+
+    def process(self):
+
+        if not self.outputs[0].is_linked:
+            return
+
+        var_names = self.get_variables()
+        inputs = self.get_input()
+        if not inputs:
+            if not var_names:
+                inputs = {'a': [0.0]}
+
+        result_vertices = []
+        result_edges = []
+        result_faces = []
+
+        template = self.load_json()
+
+        input_values = [inputs[name] for name in var_names]
+        parameters = match_long_repeat(input_values)
+        for values in zip(*parameters):
+            variables = dict(zip(var_names, values))
+
+            json = evaluate(template, variables)
+            result_vertices.append(json['vertices'])
+            result_edges.append(json['edges'])
+            result_faces.append(json['faces'])
+
+        if self.outputs['Vertices'].is_linked:
```

Comment:
might as well drop the the `is_linked tests here, because the data are already generated.